src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -2324,7 +2324,32 @@ private void filterExistingFiles() throws SnowflakeSQLException {
 
       logger.debug("received object summaries from remote storage");
 
-      for (StorageObjectSummary obj : objectSummaries) {
+      Iterator<StorageObjectSummary> objectSummariesIterator;
+      try {
+        objectSummariesIterator = objectSummaries.iterator();
+      } catch (Exception ex) {
+        logger.debug("Failed to get objectSummariesIterator. Exception: {}", ex.getMessage());
+        StringWriter sw = new StringWriter();
+        PrintWriter pw = new PrintWriter(sw);
+        ex.printStackTrace(pw);
+        logger.debug("objectSummariesIterator exception stacktrace: {}", pw.toString());
```

Comment:
Hi @sfc-gh-mknister , is this acceptable to print the stack trace to logs?

Reply:
Good question! Could you double-check with @sfc-gh-hchaturvedi ?

Since the logging is only local to customers, I think it should be fine to print everything, but if we do end up needing to mask the sensitive stuff like error message/reason from the stack trace, there's a mask function for telemetry that can probably be repurposed: [https://github.com/snowflakedb/snowflake-jdbc/blob/master/src/main/java/net/snowflake/client/jdbc/SnowflakeSQLLoggedException.java#L96-L113](https://github.com/snowflakedb/snowflake-jdbc/blob/master/src/main/java/net/snowflake/client/jdbc/SnowflakeSQLLoggedException.java#L96-L113).

Reply:
Can one of you confirm that this goes through the secret detector?

Reply:
Yes, confirmed. Just double checked it again.